test/e2e/accounts/common.ts

```diff
@@ -202,9 +202,35 @@ export async function connectAccountToTestDapp(driver: Driver) {
 export async function disconnectFromTestDapp(driver: Driver) {
   await driver.switchToWindowWithTitle(WINDOW_TITLES.ExtensionInFullScreenView);
   await driver.clickElement('[data-testid="account-options-menu-button"]');
-  await driver.clickElement('[data-testid="global-menu-connected-sites"]');
-  await driver.clickElement({ text: 'Disconnect', tag: 'a' });
-  await driver.clickElement({ text: 'Disconnect', tag: 'button' });
+
+  if (process.env.MULTICHAIN) {
+    await driver.clickElement({ text: 'All Permissions', tag: 'div' });
+    await driver.clickElement({ text: 'Got it', tag: 'button' });
+    await driver.clickElement({
+      text: '127.0.0.1:8080',
+      tag: 'p',
+    });
+    await driver.clickElement('[data-testid="account-list-item-menu-button"]');
+    await driver.clickElement({ text: 'Disconnect', tag: 'button' });
+    await driver.clickElement('[data-testid="disconnect-all"]');
+    await driver.clickElement('button[aria-label="Back"]');
+    await driver.clickElement('button[aria-label="Back"]');
+    // Validate dapp is not connected
+    await driver.clickElement('[data-testid="account-options-menu-button"]');
+    await driver.clickElement({ text: 'All Permissions', tag: 'div' });
+    await driver.assertElementNotPresent('p[data-testid="sites-connections"]');
+  } else {
+    await driver.clickElement('[data-testid="global-menu-connected-sites"]');
+    await driver.clickElement({ text: 'Disconnect', tag: 'a' });
+    await driver.clickElement({ text: 'Disconnect', tag: 'button' });
+    // Validate dapp is not connected
+    await driver.clickElement('[data-testid="account-options-menu-button"]');
+    await driver.clickElement('[data-testid="global-menu-connected-sites"]');
+    await driver.findElement({
+      text: 'Account 1 is not connected to any sites.',
+      tag: 'p',
```

Comment:
I recommend avoiding large if-else structures as they can make the code cumbersome. Instead, consider dividing the logic into two distinct functions: disconnectFromTestDappMultichain and disconnectFromTestDapp. By defining these functions separately and calling them directly where needed, we can streamline the code, enhancing its readability and ease of maintenance.